r/R/schema.R

```diff
@@ -157,6 +165,9 @@ Schema$create <- function(...) {
   .list <- list2(...)
   if (all(map_lgl(.list, ~ inherits(., "Field")))) {
     schema_(.list)
+  } else if (is.null(names(.list))) {
+    # If not supplied as fields, only other alternative is e.g.`a = int16()`
+    abort("Schema definitions must be supplied as field/data type or field name/data type pairs")
   } else {
     schema_(.fields(.list))
   }
```

Comment:
Would it be possible (and if so wise?) to check if `length(.list) == 1 && schema_check(.list[[1]])` then  schema_(.fields(.list[[1]]))? 

One could always call `do.call(schema, list)`, but it might be nice to also accept a single list containing the specifications too?

Reply:
Sure, though would we want to also expand it to check if `length(.list) == 1` && `all(map_lgl(.list, ~ inherits(.x, "DataType")))` and allow that too?

Reply:
(i.e. so the user can supply a list of name/value pairs and it's not important which of the 2 methods they use - fields or just field names)

Reply:
OH, yes absolutely (if it's not too much scope creep here!) that would be quite nice